index.js

```diff
@@ -1,39 +1,604 @@
+      simplifyString(answer) === "attend class"
+    ) {
+      if (currentState === "ON FIRE!!!") {
+        console.log(
+          "If you sit down on this nice chair while you're on fire, you'll ruin the chair. You can't bring yourself to be that person."
+        );
+        turnTaken = 0;
+      } else if (classTaken === 1) {
+        console.log(
+          "Class already happened. There's not much point in sitting down."
+        );
+        turnTaken = 0;
+      } else {
+        console.log(hungryMessage);
+        classTaken = 1;
+        enterState("Hungry");
+      }
+    } else if (
+      simplifyString(answer).includes("leave") ||
+      simplifyString(answer).includes("exit") ||
+      simplifyString(answer) === "go foyer" ||
+      simplifyString(answer) === "enter foyer"
+    ) {
+      enterRoom("Foyer");
+      console.log("You head back out into the foyer.");
+    } else if (simplifyString(answer) === "talk to josh") {
+      console.log("Josh says 'Hi.'");
+    } else {
+      console.log("Sorry, I don't know how to \"" + answer + '".');
+      turnTaken = 0;
+    }
+  } else if (
+    simplifyString(answer) === "go bathroom" ||
+    simplifyString(answer) === "enter bathroom"
+  ) {
+    console.log("The bathroom is outside the classroom, in the foyer.");
+    turnTaken = 0;
+  } else {
+    console.log("Sorry, I don't know how to \"" + answer + '".');
+    turnTaken = 0;
+  }
+  if (turnTaken === 1 && currentState === "Hungry") {
+    hungryTurnsLeft--;
+    if (hungryTurnsLeft > 3) {
+      console.log("Your stomach grumbles. Pizza is on your mind.");
+    } else if (hungryTurnsLeft > 0) {
+      console.log("Stomach...So empty...Must...Get pizza...");
+    } else if (hungryTurnsLeft === 0) {
+      console.log("You couldn't eat the pizza in time...");
+      enterState("DEAD!");
+    }
+  }
+  if (turnTaken === 1 && currentState === "ON FIRE!!!") {
+    hp = Math.round(hp - 20);
+    console.log(
+      "OOF, ouch, being on fire hurts!! You know have " +
+        hp +
+        " health points remaining!"
+    );
+  }
+  if (hp <= 0) {
+    enterState("DEAD!");
+  }
+  if (currentState === "DEAD!") {
+    console.log("YOU DIED");
   }
 }
 
 start();
 
 async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
+  console.log(mainStMessage);
+  answer = await ask(">");
+  while (true) {
+    respond(answer);
+    if (locations[currentLocation].hasBeen === 0) {
+      console.log(locations[currentLocation].welcomeMessage);
+      locations[currentLocation].hasBeen = 1;
+    }
+    if (currentState !== "DEAD!" && currentState !== "Enlightened") {
+      answer = await ask(">");
+    } else {
+      break;
+    }
+  }
   process.exit();
```

Comment:
This program looks great. Im loving the player status mechanic along with the comedic dialog. Keep up the good work